source/includes/read/change-streams.c

```diff
@@ -0,0 +1,98 @@
+#include <bson/bson.h>
+#include <mongoc/mongoc.h>
+#include <stdio.h>
+
+int
+main (int argc, char *argv[])
+{
+    mongoc_client_t *client;
+    mongoc_collection_t *collection;
+    mongoc_init ();
+
+    client =
+        mongoc_client_new ("<connection string URI>");
+    collection = mongoc_client_get_collection (client, "sample_restaurants", "restaurants");
+
+    {
+        // Opens a change stream on the collection and prints each change
+        // start-open-change-stream
+        bson_t *pipeline = bson_new ();
+        // Change stream will wait 1 minute for changes
+        bson_t *opts = BCON_NEW ("maxAwaitTimeMS", BCON_INT64 (60000));
+        const bson_t *doc;
+
+        mongoc_change_stream_t *change_stream =
+            mongoc_collection_watch (collection, pipeline, opts);
+
+        while (mongoc_change_stream_next (change_stream, &doc)) {
```

Comment:
`mongoc_change_stream_next` only returns true when a change is returned within the `maxAwaitTimeMS`. If no change was returned, and no error occurred, a later call to `mongoc_change_stream_next` can still return a change.

If the examples are intended to show the behavior of "wait (possibly forever) until there is an event or error", then I suggest looping as follows:

```c
while (true)
{
  bson_error_t error;
  if (mongoc_change_stream_next(change_stream, &doc))
  {
    char *str = bson_as_canonical_extended_json(doc, NULL);
    printf("Received change: %s\n", str);
    bson_free(str);
  }
  else if (mongoc_change_stream_error_document(change_stream, &error, NULL))
  {
    printf("Got error on change stream: %s\n", error.message);
    break;
  }
}
```